### PR TITLE
Bring tests to 100% code coverage

### PR DESCRIPTION
**Pull request by DeviaVir**
```
Name                                            Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------
spacel.py                                           0      0   100%
spacel/args.py                                     67      0   100%
spacel/aws.py                                       3      0   100%
spacel/aws/ami.py                                  22      0   100%
spacel/aws/clients.py                              37      0   100%
spacel/aws/instance_types.py                        1      0   100%
spacel/model.py                                     2      0   100%
spacel/model/app.py                                87      0   100%
spacel/model/orbit.py                              74      0   100%
spacel/provision.py                                17      0   100%
spacel/provision/alarm.py                           3      0   100%
spacel/provision/alarm/actions.py                   6      0   100%
spacel/provision/alarm/endpoint.py                  0      0   100%
spacel/provision/alarm/endpoint/email.py           18      0   100%
spacel/provision/alarm/endpoint/factory.py         29      0   100%
spacel/provision/alarm/endpoint/pagerduty.py      114      0   100%
spacel/provision/alarm/endpoint/scale.py           30      0   100%
spacel/provision/alarm/endpoint/slack.py           26      0   100%
spacel/provision/alarm/factory.py                  19      0   100%
spacel/provision/alarm/trigger.py                   0      0   100%
spacel/provision/alarm/trigger/factory.py         118      0   100%
spacel/provision/alarm/trigger/metrics.py          47      0   100%
spacel/provision/changesets.py                     27      0   100%
spacel/provision/cloudformation.py                155      0   100%
spacel/provision/db.py                              2      0   100%
spacel/provision/db/base.py                        12      0   100%
spacel/provision/db/cache.py                       53      0   100%
spacel/provision/db/rds.py                        116      0   100%
spacel/provision/orbit.py                           3      0   100%
spacel/provision/orbit/gdh.py                      80      0   100%
spacel/provision/orbit/provider.py                 25      0   100%
spacel/provision/orbit/space.py                   109      0   100%
spacel/provision/provision.py                      20      0   100%
spacel/provision/s3.py                              2      0   100%
spacel/provision/s3/base.py                        16      0   100%
spacel/provision/s3/lambda_uploader.py             36      0   100%
spacel/provision/s3/template_uploader.py            8      0   100%
spacel/provision/template.py                        6      0   100%
spacel/provision/template/app.py                  125      0   100%
spacel/provision/template/app_spot.py              57      0   100%
spacel/provision/template/base.py                  42      0   100%
spacel/provision/template/bastion.py               30      0   100%
spacel/provision/template/ingress_resource.py      52      0   100%
spacel/provision/template/tables.py                 9      0   100%
spacel/provision/template/vpc.py                   88      0   100%
spacel/security.py                                  4      0   100%
spacel/security/acm.py                             44      0   100%
spacel/security/kms_crypt.py                       54      0   100%
spacel/security/kms_key.py                         40      0   100%
spacel/security/password.py                        55      0   100%
spacel/security/payload.py                         18      0   100%
spacel/user.py                                      1      0   100%
spacel/user/ssh_db.py                              49      0   100%
-----------------------------------------------------------------------------
TOTAL                                            2058      0   100%
----------------------------------------------------------------------
Ran 245 tests in 0.594s

OK
```

### CODE REVIEW

**Comment by DeviaVir** [comment]
I see that [jenkins](https://jenkins.pebble.com/job/pebble/job/spacel-provision/job/full_tests/2/console) notices a lot more missing still, will update.

**Comment by DeviaVir** [comment]
Updated, [jenkins](https://jenkins.pebble.com/job/pebble/job/spacel-provision/job/full_tests/3/console) agrees.

File: src/test/model/test_app.py [hunk 1]
```diff
@@ -71,6 +71,38 @@ def test_services_docker(self):
...
+        app = SpaceApp(self.orbit, {
+            'services': {
+                SERVICE_NAME: {
+                    'unit_file': '[Unit]'
+                }
+            }
+        })
+
+        self.assertEqual(1, len(app.services))
+        self.assertIsNotNone(app.services[SERVICE_NAME].unit_file)
```
Comment (thepwagner) [inline src/test/model/test_app.py]: Seems repetitive given `assertEquals()` below. Meh.

File: src/test/provision/orbit/test_gdh.py [hunk 2]
```diff
@@ -45,11 +45,18 @@ def setUp(self):
...
+                'OutputKey': 'SecurityGroup',
+                'OutputValue': 'sg-000001'
+            }]
+        }
 
         self.orbit_factory.get_orbit(self.orbit)
 
         self.cloudformation.describe_stack_resource.assert_called_once()
         self.orbit_factory._orbit_from_child.assert_called_once()
+        self.assertEqual(self.orbit._bastion_sgs[REGION], 'sg-000001')
```
Comment (thepwagner) [inline src/test/provision/orbit/test_gdh.py]: Nit: calling the "public" function `bastion_sg(REGION)` is better practice than accessing the "private" field directly.

File: src/test/provision/orbit/test_space.py [hunk 3]
```diff
@@ -157,6 +159,8 @@ def test_orbit_from_vpc(self):
         self.assertEquals(['subnet-000111'],
                           self.orbit.public_elb_subnets(REGION))
         self.assertEquals(VPC_ID, self.orbit.vpc_id(REGION))
+        self.assertEquals(SPOTFLEET_ROLE_ARN,
+                          self.orbit._spot_fleet_role[REGION])
```
Comment (thepwagner) [inline src/test/provision/orbit/test_space.py]: Nit: same public/private access thing.

**Pull request merged**